### PR TITLE
InvocationTargetException in MetricConsumer, requiring consumer restart

### PR DESCRIPTION
**Pull request by johnlhamilton**
Added additional error logging on public web socket and api methods to make it easier to track down this problem the next time it occurs.

ZEN-19869

### CODE REVIEW

File: metric-consumer-app/src/main/java/org/zenoss/app/consumer/metric/remote/MetricWebSocket.java [hunk 1]
```diff
@@ -104,7 +110,8 @@ public Control onMessage(byte[] data, WebSocketSession session) {
                 log.error("Invalid message");
                 return Control.malformedRequest("Invalid message");
             }
-        } catch (RuntimeException e) {
+        } catch (Exception e) {
```
Comment (iancmcc) [inline metric-consumer-app/src/main/java/org/zenoss/app/consumer/metric/remote/MetricWebSocket.java]: Shouldn't we re-throw in this case?

File: metric-consumer-app/src/main/java/org/zenoss/app/consumer/metric/remote/MetricWebResource.java [hunk 2]
```diff
@@ -65,29 +65,35 @@ public MetricWebResource(MetricService metricService, ZappSecurity security, Con
...
+                log.debug("Tagging metrics with tenant-id: {}", tenant.id());
+                Utils.injectTag("zenoss_tenant_id", tenant.id(), metrics);
+            }
+
+            //filter tags using configuration white list
+            Utils.filterMetricTags(metrics, configuration.getTagWhiteList());
+
+            String remoteIp = Utils.remoteAddress(request);
+            return metricService.push(metrics, remoteIp, null);
+        } catch (Exception e) {
```
Comment (iancmcc) [inline metric-consumer-app/src/main/java/org/zenoss/app/consumer/metric/remote/MetricWebResource.java]: Should be able to catch RuntimeException, although not sure if that provides any benefit.

File: metric-consumer-app/src/main/java/org/zenoss/app/consumer/metric/remote/MetricWebResource.java [hunk 3]
```diff
@@ -65,29 +65,35 @@ public MetricWebResource(MetricService metricService, ZappSecurity security, Con
...
+            log.info("post(metricCollection={}, request={}", metricCollection, request);
+            log.error("Unexpected exception: " + e.getMessage(), e);
+            return Control.error(e.getMessage());
```
Comment (iancmcc) [inline metric-consumer-app/src/main/java/org/zenoss/app/consumer/metric/remote/MetricWebResource.java]: return Control.error or re-throw?

File: metric-consumer-app/src/main/java/org/zenoss/app/consumer/metric/remote/MetricWebResource.java (hunk 2 above)
Comment (johnlhamilton) [inline metric-consumer-app/src/main/java/org/zenoss/app/consumer/metric/remote/MetricWebResource.java]: I'm going to change this to just log and re-throw, so I don't think I would handle a RTE any differently than a plain Exception.

**Pull request merged**